fix: Correct user prop type from null to undefined for MainNavbar

@claude
Fix Prompt:
markdown@claude Please fix the TypeScript error TS2322 that is causing the Vercel deployment to fail.

Build Error:
src/App.tsx(24,9): error TS2322: Type 'NavbarUserType | null' is not assignable to type 'UserType | undefined'.

Issue:
The useUserStore provides a user state that is NavbarUserType | null. The MainNavbar component's user prop expects a type of UserType | undefined. The build is failing because null cannot be assigned to a prop that expects undefined.

Required Fix:

In src/App.tsx, locate the line where the <MainNavbar /> component is rendered.

Modify the user prop being passed to it. Convert the user state from the store so that if it is null, it becomes undefined.

Example Fix:

Change this line:

TypeScript

<MainNavbar user={user} ... />
To this:

TypeScript

<MainNavbar user={user || undefined} ... />
Verification:
Ensure the change resolves the type error and allows the command npm run build to complete successfully.

--- a/src/App.tsx
+++ b/src/App.tsx
@@ -21,7 +21,7 @@ function App() {
     <div className="min-h-screen bg-parchment flex flex-col">
       {/* Main Navigation */}
       <MainNavbar 
-        user={user}
+        user={user || undefined}
         onNavigate={(page) => console.log('Navigate to:', page)}
         onLogout={() => console.log('Logout')}
         notificationCount={3}

